source/structured_data.cc

```diff
@@ -604,6 +607,174 @@ namespace aspect
     }
 
 
+
+    template <int dim>
+    void
+    StructuredDataLookup<dim>::load_netcdf(const std::string &filename)
+    {
+#ifndef ASPECT_WITH_NETCDF
+      (void)filename;
+      AssertThrow(false, ExcMessage("Loading NetCDF files is only supported if ASPECT is configured with the NetCDF library!"));
+#else
+      TableIndices<dim> new_table_points;
+      std::vector<std::string> coordinate_column_names(dim);
+      std::vector<std::string> data_column_names;
+      std::vector<Table<dim,double>> data_tables;
+      std::vector<std::vector<double>> coordinate_values(dim);
+
+      int ncid;
+      int status;
+
+      status = nc_open(filename.c_str(), NC_NOWRITE, &ncid);
+      AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+      int ndims, nvars, ngatts, unlimdimid;
+      status = nc_inq(ncid, &ndims, &nvars, &ngatts, &unlimdimid);
+      AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+
+
+      for (int dimid=0; dimid<ndims; ++dimid)
+        {
+          char  name[NC_MAX_NAME];
+          size_t length;
+          status = nc_inq_dim(ncid, dimid, name, &length);
+          AssertThrow(status == NC_NOERR, ExcMessage("NetCDF Error occured!"));
+          std::cout << "dim " << dimid << "'" << name << "' has length " << length << std::endl;
+          // TODO: parse/store coordinate name?
```

Comment:
What do we do with the table headings for ASCII data?

Reply:
for the data columns: we use them, for the coordinates: discard at the moment